src/Features/Core/Portable/FindReferences/IDefinitionsAndReferencesFactory.cs

```diff
@@ -0,0 +1,273 @@
+                case SymbolKind.NamedType:
+                case SymbolKind.PointerType:
+                    return 1;
+
+                default:
+                    return 2;
+            }
+        }
+
+        private void ProcessReferencedSymbol(
+            Solution solution,
+            ReferencedSymbol referencedSymbol,
+            ImmutableArray<DefinitionItem>.Builder definitions,
+            ImmutableArray<SourceReferenceItem>.Builder references,
+            HashSet<ValueTuple<Document, TextSpan>> uniqueLocations)
+        {
+            // See if this is a symbol we even want to present to the user.  If not,
+            // ignore it entirely (including all its reference locations).
+            if (!referencedSymbol.ShouldShow())
+            {
+                return;
+            }
+
+            // Try to create an item for this definition.  If we can't,
+            // ignorei it entirely (including all its reference locations).
+            var definitionItem = CreateDefinitionItem(solution, referencedSymbol);
+            if (definitionItem == null)
+            {
+                return;
+            }
+
+            definitions.Add(definitionItem);
+
+            // Now, create the SourceReferenceItems for all the reference locations
+            // for this definition.
+            CreateReferences(referencedSymbol, references, definitionItem, uniqueLocations);
+
+            // Finally, see if there are any third parties that want to add their
+            // own result to our collection.
+            var thirdPartyItem = GetThirdPartyDefinitionItem(solution, referencedSymbol.Definition);
+            if (thirdPartyItem != null)
+            {
+                definitions.Add(thirdPartyItem);
+            }
+        }
+
+        /// <summary>
+        /// Provides an extension point that allows for other workspace layers to add additional
+        /// results to the results found by the FindReferences engine.
+        /// </summary>
+        protected virtual DefinitionItem GetThirdPartyDefinitionItem(
+            Solution solution, ISymbol definition)
+        {
+            return null;
+        }
+
+        private static DefinitionItem CreateDefinitionItem(
+            Solution solution, ReferencedSymbol referencedSymbol)
+        {
+            var definition = referencedSymbol.Definition;
+
+            // First, determine the set of locations for this symbol that we'll
+            // want to present to the user.  Some symbols (like namespace) may
+            // have many locations, but we'll only want to show the user a single
+            // one.
+            var filteredLocations = FilterDefinitionLocations(definition);
+
+            // Convert the filtered set of locations to DefinitionLocation instances.
+            // If weren't any to, say because the symbol had no locations that we
+            // could understand (like Location.None), then we skip this item.
```

Comment:
But I don't actually see how this comment applies. No skipping is happening here.